File: 300. Longest Increasing Subsequence.py
class Solution(object):
    def lengthOfLIS(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 0:
            return 0
        T = [1 for i in xrange(len(nums))]
        res = 1
        for i in xrange(1, len(nums)):
            for j in xrange(i):
                if nums[i] > nums[j]:
                    T[i] = max(T[i], T[j] + 1)
            res = max(res, T[i])
        return res

File: 300. Longest Increasing Subsequence.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        vector <int> T (nums.size(), 1);
        int res = 1;
        for (int i = 1; i < nums.size(); i++)
        {
            for (int j = 0; j < i; j++)
            {
                if (nums[i] > nums[j] && T[j] + 1 > T[i])
                    T[i] = T[j] + 1;
            }
            res = T[i] > res ? T[i] : res;
        }
        return res;
    }
};
